### PR TITLE
use retry after instead of calculating delay

### PR DESCRIPTION
**Pull request by ndurell**
# Description

# Check List

- [ ] Are you changing anything with the public API?
- [ ] Have you tested this change on real device?
- [ ] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compatible with all the operating system version this SDK currently supports?

# Manual Test Plan

1.

# Supporting Materials

### CODE REVIEW

File: Sources/KlaviyoSwift/APIRequestErrorHandling.swift [hunk 1]
```diff
@@ -112,7 +112,7 @@ func handleRequestError(
         case let .retryWithBackoff(requestCount, totalCount, _):
             requestRetryCount = requestCount + 1
             totalRetryCount = totalCount + 1
-            nextBackoff = getDelaySeconds(for: totalRetryCount)
+            nextBackoff = retryAfter ?? getDelaySeconds(for: totalRetryCount)
```
Comment (evan-masseau) [inline Sources/KlaviyoSwift/APIRequestErrorHandling.swift:115]: Should we apply jitter to the `retryAfter` too?

Comment (evan-masseau) [reply Sources/KlaviyoSwift/APIRequestErrorHandling.swift:115]: I suppose if we needed jitter, the server could apply it.

File: Sources/KlaviyoSwift/KlaviyoAPI.swift [hunk 2]
```diff
@@ -70,7 +70,8 @@ struct KlaviyoAPI {
 
         if httpResponse.statusCode == 429 {
             requestRateLimited(request)
-            return .failure(KlaviyoAPIError.rateLimitError)
+            let retryAfter = httpResponse.allHeaderFields["Retry-After"] as? Int
```
Comment (evan-masseau) [inline Sources/KlaviyoSwift/KlaviyoAPI.swift:73]: may not affect ios, but on android this response headers' value was a map of lists of strings, `Map<String, List<String>>`

File: Sources/KlaviyoSwift/APIRequestErrorHandling.swift (hunk 1 above)
Comment (evan-masseau) [reply Sources/KlaviyoSwift/APIRequestErrorHandling.swift:115]: @jason-myers-klaviyo do you have any opinion on whether the SDKs should apply a jitter on top of the `Retry-After` value? Does the `Retry-After` just represent the time till the steady state rate limit "resets" or is there more advanced logic to it?

Comment (jason-myers-klaviyo) [reply Sources/KlaviyoSwift/APIRequestErrorHandling.swift:115]: Jitter longer than the seconds given would be fine and up to you.  Nice to have I think

Jitter shorter will cause another 429 I think

Comment (jason-myers-klaviyo) [reply Sources/KlaviyoSwift/APIRequestErrorHandling.swift:115]: It is just the time until the steady windows reset IIRC

Comment (evan-masseau) [reply Sources/KlaviyoSwift/APIRequestErrorHandling.swift:115]: Oh ya definitely only _add_ jitter

**Review by evan-masseau — CHANGES_REQUESTED** [review]
Add jitter (1-10s?) to server's `Retry-After` value

**Comment by ajaysubra** [comment]
> Add jitter (1-10s?) to server's `Retry-After` value

Added jitter to `Retry-After`. Will test it out in the morning.

**Review by ajaysubra — APPROVED** [review]

File: Sources/KlaviyoSwift/KlaviyoAPI.swift (hunk 2 above)
Comment (evan-masseau) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift:73]: Were you able to test that this way to parse `Retry-After` header does work on live 429 response from production?

**Review by evan-masseau — APPROVED** [review]
LGTM, just confirming if its been tested against a live 429 response from production

File: Sources/KlaviyoSwift/KlaviyoAPI.swift (hunk 2 above)
Comment (ajaysubra) [reply Sources/KlaviyoSwift/KlaviyoAPI.swift]: I didn't have a chance to test it out last night but tested now and it was indeed not returning the header value. Switched to using `value(forHTTPHeaderField:` which works.

**Pull request merged**